### PR TITLE
Using numpy.random.RandomState instead of numpy.random in LDA

### PR DESCRIPTION
**Pull request by droudy**
In reference to #113 , a random_state parameter has been added to the LdaState initializer and a check_random_seed method as well to check the seed. The check_random_state method was taken from https://github.com/maciejkula/glove-python/pull/32/commits/95a04af9ef76c1b13949dc92c4a1bb1ecaa5cf3c , should I rewrite the method?

### CODE REVIEW

**Comment by tmylk** [comment]
Please add tests for this particular function(checking both initializaitons work).

Also need to update existing tests that fix random seed to use this new API, for example https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/test/test_ldamodel.py#L253

**Comment by tmylk** [comment]
Just need to add `random_state` to MultiCore

File: gensim/models/hdpmodel.py [hunk 1]
```diff
@@ -586,6 +586,7 @@ def show_topics(self, num_topics=10, num_words=10, log=False, formatted=True):
         if num_topics < 0:
             num_topics = len(self.data)
 
+
```
Comment (tmylk) [inline gensim/models/hdpmodel.py]: remove the line

File: gensim/models/ldamodel.py [hunk 2]
```diff
@@ -91,6 +91,19 @@ def update_dir_prior(prior, N, logphat, rho):
 
     return prior
 
+def check_random_state(seed):
```
Comment (tmylk) [inline gensim/models/ldamodel.py]: `get_random_state` is a better name

File: gensim/test/test_ldamodel.py [hunk 3]
```diff
@@ -48,6 +48,11 @@ def testfile():
     return os.path.join(tempfile.gettempdir(), 'gensim_models.tst')
 
 
+def testRandomState():
+    testcases = [numpy.random.seed(0), None, numpy.random.RandomState(0), 0]
+    for testcase in testcases:
+        assert(isinstance(ldamodel.check_random_state(testcase), numpy.random.RandomState))
```
Comment (tmylk) [inline gensim/test/test_ldamodel.py]: add a test `assertEqual(ldamodel.check_random_state(testcase), numpy.random.RandomState(0))`

File: gensim/test/test_ldamodel.py [hunk 4]
```diff
@@ -97,7 +98,7 @@ def testAlpha(self):
         expected_shape = (2,)
 
         # should not raise anything
-        self.class_(**kwargs)
+        self.class_(**kwarghs)
```
Comment (piskvorky) [inline gensim/test/test_ldamodel.py]: @tmylk what is this change?

**Comment by droudy** [comment]
@piskvorky it was an accident, it was fixed in https://github.com/RaRe-Technologies/gensim/pull/748/commits/8ea4d5852163b0c0178994336a3778aecbe30bf1

**Pull request merged**